db/migrate/20200114164456_normalize_ace_table.rb

```diff
@@ -0,0 +1,33 @@
+class NormalizeAceTable < ActiveRecord::Migration[5.2]
+  def up
+    create_table :permissions do |t|
+      t.integer :name
+
+      t.timestamps
+    end
+
+    create_join_table :access_control_entries, :permissions do |t|
+      t.index [:access_control_entry_id, :permission_id], :name => "index_ace_permissions_on_ace_id_and_permission_id"
+    end
+
+    (0..3).each do |permission|
+      Permission.create!(:name => permission)
+    end
+
+    AccessControlEntry.all.each do |entry|
+      perm = entry.send(:permission)
+      permission = Permission.find_by(:name => perm)
+      entry.permissions << permission
```

Comment:
@syncrou 
If the table today looks like

Guid      Permission   ID   ObjectType
guid1      read            20    Portfolio
guid1      write           20    Portfolio
guid2     order           30    Portfolio

After the migration it should have 

Guid      ID   ObjectType
guid1     20    Portfolio
guid2     30    Portfolio

This should be deleting rows from the AcessControlEntry table, the migration seems to keep the ACE's in place and just add the permissions relationship.
Should this be done in multiple passes where we would 
(1) collect all the group_uuids and their associated permissions   e.g
      [{'guid1' => {'object_id' => 20, :permissions => ['read', 'update']}, 
       {'guid2' => {'object_id' => 30, :permissions => ['delete']}]
(2) Delete all the rows
(3) Recreate the rows with just the collected data